cs/testing-with-travis.texy

```diff
@@ -0,0 +1,245 @@
+Průběžné testování s Nette Tester
+********************************
+
+/--div .[perex]
+Znáte to. Napsali jste testy, které teď procházejí, poplácali se po zádech a váš spánek začal být mnohem klidnější. Máte jistotu, že váš kód spolehlivě funguje. A pak se to stane. Hodiny ukazují 3:15 a do nočního ticha se zakousne nepříjemný zvuk drnčení mobilu na nočním stolku. Volá šéf. Zase to nefunguje. Zalije vás studený pot. Vždyť na to máme testy! V tu chvíli se vynoří vzpomínka na páteční odpoledne - spěcháte domů a zapomínáte spustit testy po poslední úpravě.
+
+Abyste předešli podobným zážitkům, ukážeme si, jak průběžně testovat pomocí populárního nástroje [Travis CI](https://travis-ci.org/):
+
+- nastavení testovacího prostředí
+- instalace závislostí pomocí Composeru
+- spouštění testů pomocí Nette/Tester
+- nastavení dalších služeb
+- [TL;DR](#Výsledný soubor)
+- integrace s GitHubem
+\--
+
+Jak funguje Travis CI
+=====================
+
+Travis CI((Continuous Integration)) (dále jen Travis) je velmi populární služba pro automatizaci [průběžné integrace](http://cs.wikipedia.org/wiki/Průběžná_integrace). Je poskytována zdarma((Pro soukromé repozitáře je zpoplatněna.)) a nabízí automatizované spouštění testů po zápisu změny do repozitáře. Je plně integrována do prostředí GitHubu.
+
+Travis se konfiguruje souborem `.travis.yml` umístěným v kořenovém adresáři projektu. Obsahuje několik bloků, které si podrobněji popíšeme.
+
+Bloky jsou spouštěny v pořadí 
+- `before_install`
+- `install`
+- `before_script`
+- `script`
+- `after_success` nebo `after_failure`
+- `after_script`
+
+.[warning]
+Soubor `.travis.yml` dovoluje odsazovat pouze mezerami. Syntaktickou správnost si můžete ověřit [pomocí validátoru |http://lint.travis-ci.org/].
+
+Pro podrobnější informace si přečtěte [oficiální dokumentaci |http://docs.travis-ci.com].
+
+Vytvoření .travis.yml
+=====================
+
+Nastavení jazyka
+----------------
+
+Nejprve je potřeba pomocí `language` Travis informovat, pro jaký jazyk má prostředí nastavit. Na dalších řádcích uvádíme, pro jaké verze PHP budeme chtít náš projekt otestovat. Testy pak budou spuštěny pro každou verzi zvlášť. Nette Tester podporuje i [HHVM](http://hhvm.com), můžeme ho tedy přidat také.
+
+/--
+language: php
+
+php:
+  - 5.3.3
+  - 5.4
+  - 5.5
+  - 5.6
+  - hhvm
+\--
+ 
+Proměnné prostředí
+------------------
+
+V sekci `env` nastavujeme další varianty prostředí pro běh testů. Pro každou odrážku se skripty provedou samostatně a v jedné odrážce lze definovat více proměnných. Dále v textu budeme používat proměnnou `TESTER_PHP_BIN`, abychom pro HHVM mohli testy pouštět s parametrem `-p hhvm`.
+
+/--
+env:
+  - TESTER_PHP_BIN="php-cgi"
+  - TESTER_PHP_BIN="hhvm"
+\--
+
+Uvedené sekce php a env nám zajistí spuštění testů celkem 10x. Pro pět verzí PHP a dvě varianty proměnné `TESTER_PHP_BIN`.
+
+Instalace závislostí
+--------------------
+
+.[caution]
```

Comment:
Radši `.[note]`, pro většinu lidí tohle není podstatná informace, takže netřeba strašit vykřičníkem.